hudi-common/src/main/java/org/apache/hudi/common/model/OverwriteWithLatestAvroPayload.java

```diff
@@ -67,7 +74,8 @@ public OverwriteWithLatestAvroPayload preCombine(OverwriteWithLatestAvroPayload
 
     GenericRecord genericRecord = (GenericRecord) recordOption.get();
     // combining strategy here trivially ignores currentValue on disk and writes this record
-    Object deleteMarker = genericRecord.get("_hoodie_is_deleted");
+    String deleteField = userDefineDeleteField == null ? DEFAULT_DELETE_FIELD : userDefineDeleteField;
+    Object deleteMarker = genericRecord.get(deleteField);
```

Comment:
The only caller to this method is `org.apache.hudi.io.HoodieMergeHandle#write`, where you have access to `HoodieWriteConfig config`. 

If you make the configuration `delete.field` accessible from `HoodieWriteConfig` then make this method `combineAndGetUpdateValue(IndexedRecord currentValue, Schema schema, Option<String> deleteField)`, the overall logic will be simiplified a lot.

Reply:
To maintain full compatibility, just provide an overloading version like
```java
combineAndGetUpdateValue(IndexedRecord currentValue, Schema schema) {
  combineAndGetUpdateValue(currentValue, schema, Option.emtpy())
}
```

Reply:
It sounds good. But do we need to add the new method to the interface HoodieRecordPayload? If so, does the classes that  has implemented need to implement this method?

Reply:
> do we need to add the new method to the interface HoodieRecordPayload

hmmm. that's tricky. Sorry i did not realize this is an interface API.. does not look clean to make it into the interface... then keeping it local to the class lose agnosticness when calling `combineAndGetUpdateValue()`.. need more thoughts on this..

Reply:
@shenh062326 Maybe we shouldn't do the check in the payload class itself. Maybe `org.apache.hudi.io.HoodieMergeHandle#write` is better for this job. After getting `Option<IndexedRecord> combinedAvroRecord`, we check it against the delete field defined in the configs and convert it to Option.empty() if appropriate. As this feature is config-related, whoever owns the configs should do it. Need more inputs from @nsivabalan